### PR TITLE
Move more tests out of `sql.spec.ts`

### PR DESCRIPTION
**Pull request by samwho**
## Description

This continues the work of https://github.com/Budibase/budibase/pull/13734 in moving more and more tests out of `sql.spec.ts` and into various integration test files.

### CODE REVIEW

File: packages/server/src/api/routes/tests/search.spec.ts [hunk 1]
```diff
@@ -73,7 +74,7 @@ describe.each([
   })
 
   async function createTable(schema: TableSchema) {
-    table = await config.api.table.save(
+    return await config.api.table.save(
```
Comment (adrinr) [inline packages/server/src/api/routes/tests/search.spec.ts:77]: ❤️

File: packages/server/src/api/routes/tests/search.spec.ts [hunk 2]
```diff
@@ -1506,4 +1540,52 @@ describe.each([
         expectQuery({ equal: { "1:1:name": "none" } }).toFindNothing())
     })
   })
+
+  // This will never work for Lucene.
+  // TODO(samwho): fix for SQS
```
Comment (adrinr) [inline packages/server/src/api/routes/tests/search.spec.ts:1545]: Is this meant to be done before merging?

Comment (samwho) [reply packages/server/src/api/routes/tests/search.spec.ts:1545]: Ah, @mike12345567 actually fixed this. Lemme "uncomment" the test. Good catch!

Comment (samwho) [reply packages/server/src/api/routes/tests/search.spec.ts]: Was just the comment that needed removing, the condition was already allowing the SQS test variant to run.

**Review by mike12345567 — APPROVED** [review]
LGTM!

**Pull request merged**